python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -256,6 +262,18 @@ def tag(key: str, value: str, include_sources: bool = False) -> "AssetSelection"
         """
         return TagAssetSelection(key=key, value=value, include_sources=include_sources)
 
+    @staticmethod
+    def kind(kind: Optional[str], include_sources: bool = False) -> "AssetSelection":
+        """Returns a selection that includes materializable assets that have the provided kind, and
+        all the asset checks that target them.
+
+        Args:
+            kind (str): The kind to select.
+            include_sources (bool): If True, then include external assets matching the kind in the
+                selection.
+        """
+        return KindAssetSelection(kind_str=kind, include_sources=include_sources)
```

Comment:
I think it's possible to just represent this as `TagAssetSelection(key="<fancy kind tag key here>", value=None)`, so we can avoid adding the new class (I'm fine with keeping this static constructor for convenience)

Reply:
The thing I don't like about using `TagAssetSelection` is that on the `to_selection_str` function I have to special case that key and conditionally render `kind:<null>` or `kind:my_kind`. I do think we could maybe find a way to consolidate the implementations

Reply:
So the concrete case I'm worried about is the user writes a sensor definition targeting `kind:dbt`. Then when we retrieve the selection of the sensor the python is going to decode the serdes object as being a TagAssetSelection. That means we'll render the target as `tag:dagster/kind/dbt` which is obviously not what the user put in. 

From an engineering standpoint it also doesn't feel as maintainable to have this fork of behavior within tag when we're already forking with AssetSelection subtypes.